common/src/test/java/org/apache/comet/parquet/TestFileReader.java

```diff
@@ -609,7 +611,9 @@ public void testColumnIndexReadWrite() throws Exception {
       assertEquals(1, offsetIndex.getFirstRowIndex(1));
       assertEquals(3, offsetIndex.getFirstRowIndex(2));
 
-      assertNull(indexReader.readColumnIndex(footer.getBlocks().get(2).getColumns().get(0)));
+      if (!isSpark40Plus()) {
+        assertNull(indexReader.readColumnIndex(footer.getBlocks().get(2).getColumns().get(0)));
```

Comment:
Is this because of ANSI mode?

Reply:
In L539, it has 
```
// Creating huge stats so the column index will reach the limit and won't be written
```
This line 
```
assertNull(indexReader.readColumnIndex(footer.getBlocks().get(2).getColumns().get(0)));
```
is trying to read the column index metadata for the first column of the third row group, and verify it's null.

I am not sure why this failed for 4.0. My guess is that in the new parquet version, the column index implementation gets changed, but I didn't find the corresponding change for this.

Reply:
Do you mind tracking this in a ticket please?